mParticle-Apptentive/MPKitApptentive.m

```diff
@@ -241,10 +236,18 @@ - (MPKitExecStatus *)logCommerceEvent:(MPCommerceEvent *)commerceEvent {
 #pragma mark Events
 
 - (MPKitExecStatus *)logEvent:(MPEvent *)event {
-    BOOL success = [[Apptentive sharedConnection] engage:event.name fromViewController:nil];
+    [[Apptentive sharedConnection] engage:event.name fromViewController:nil];
 
-    MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:[[self class] kitCode] returnCode:success ? MPKitReturnCodeSuccess : MPKitReturnCodeRequirementsNotMet];
+    MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:[[self class] kitCode] returnCode:MPKitReturnCodeSuccess];
     return execStatus;
 }
 
+#pragma mark Conversation state
+
+- (void)conversationStateChangedNotification:(NSNotification *)notification {
+    MParticleUser *currentUser = [[[MParticle sharedInstance] identity] currentUser];
```

Comment:
Please access the user through the kitApi: https://github.com/mParticle/mparticle-apple-sdk/blob/master/mParticle-Apple-SDK/Kits/MPKitAPI.h#L17

It's possible in rare circumstances, if the user has not been identified yet, for the mpid to be 0. I'd recommend protecting against this case.